drs3/core/main.py

```diff
@@ -75,6 +76,32 @@ def get_drs_object_serve(
     return None
 
 
+def handle_registered_file(
+    message: Dict[str, Any],
+    publish_object_registered: Callable[[DrsObjectInitial, Config], None],
+    config: Config = CONFIG,
+):
+    """
+    Add a new entry, based on the processed message, to the database and then publish
+    a message that we did so
+    """
+
+    # we add a fictional size for testing purposes, this function is currently not used
+    drs_object = DrsObjectInitial(
+        file_id=message["file_id"],
+        md5_checksum=message["md5_checksum"],
+        registration_date=message["timestamp"],
+        size=1000,
+    )
```

Comment:
The conversion of a message to a pydantic model should happen ideally in the pubsub package. Like this, the core modules only needs to be aware of handeling these pydantic models so take a DrsObjectInitial object as input. I left a note below where you can move this code to.